config/api_client.go

```diff
@@ -49,7 +49,11 @@ func (c *Config) NewApiClient() (*httpclient.ApiClient, error) {
 				return nil
 			},
 			func(r *http.Request) error {
-				ctx := useragent.InContext(r.Context(), useragent.AuthKey, c.AuthType)
+				authType := c.AuthType
+				if t := r.Context().Value(useragent.AuthKey); t != nil {
```

Comment:
Does `go vet` show yellow squigglies for this? I think we need to define these context keys as separate types to prevent users from accidentally overriding them.